common/adapter/error.go

```diff
@@ -173,3 +174,55 @@ func ToThriftError(st *status.Status) error {
 
 	return &shared.InternalServiceError{Message: fmt.Sprintf("temporal internal uncategorized error, msg: %s", st.Message())}
 }
+
+// ToProtoError converts Thrift error to gRPC error.
+func ToServiceError(in error) error {
+	if in == nil {
+		return nil
+	}
+
+	switch thriftError := in.(type) {
+	case *shared.InternalServiceError:
+		return serviceerror.NewInternal(thriftError.Message)
+	case *shared.BadRequestError:
+		switch thriftError.Message {
+		case "No permission to do this operation.":
```

Comment:
😢

Reply:
I will convert it to proper `PermissionDenied` error.